File: ConsoleApp1/Simulation.cs
using System;
using System.Collections.Generic;
using System.Diagnostics.SymbolStore;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MTS
{
    public class Simulation
    {
        //BusSystem BusSystem = new BusSystem();

        public void CreateSimulation(StreamReader file)
        {

            Readfile(file);
            for (int i = 0; i < 20; i++)
            {
                MoveNextBusButtonClick();
            }

            //ResetBus();
        }

        public void Readfile(StreamReader file)
        {
            file.DiscardBufferedData();
            file.BaseStream.Seek(0, System.IO.SeekOrigin.Begin);
            int counter = 0;
            string line;
            while ((line = file.ReadLine()) != null)
            {
                List<string> result = line.Split(',').ToList();
                BusSystem.CreateObject(result);
                System.Console.WriteLine(line);
                counter++;
            }

        }
        public void MoveNextBusButtonClick()
        {
          BusSystem.MoveNextBus();
        }

        public void ResetBus()
        {
           // BusSystem = new BusSystem();
            //Readfile(file);
        }


        //Bus b3 = new Bus(30, 61, 0, 0, 15, 100, 300, 20);
    }
}

File: JavaPrototype/src/MTS/Simulation.java
package MTS;
import java.io.FileNotFoundException;
import java.util.Scanner;
import java.io.File;
public class Simulation {

    public void CreateSimulation(String scenarioFile) throws FileNotFoundException {
        Scanner takeCommand = new Scanner(new File(scenarioFile));
        ReadFile(takeCommand);
        for (int i = 0; (i < 20); i++) {
            MoveNextBusButtonClick();
        }

    }

    private void ReadFile(Scanner takeCommand) {
        final String DELIMITER = ",";
        String[] tokens;
        do {
            String userCommandLine = takeCommand.nextLine();
            tokens = userCommandLine.split(DELIMITER);
            BusSystem.CreateObject(tokens);


        } while (takeCommand.hasNextLine());

        takeCommand.close();
    }
    public void MoveNextBusButtonClick()
    {
        BusSystem.MoveNextBus();
    }
}
